proto/panacea/datadeal/v0/deal.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+package panacea.datadeal.v0;
+
+option go_package = "github.com/medibloc/panacea-oracle/pb/datadeal/v0";
+
+import "google/api/annotations.proto";
+
+service DataDealService {
+  rpc ValidateData(ValidateDataRequest) returns (ValidateDataResponse) {
+    option (google.api.http) = {
+      post: "/v0/data-deal/deals/{deal_id}/data"
+      body: "*"
+    };
+  }
+}
+
+message ValidateDataRequest {
+  uint64 deal_id = 1 [json_name = "deal_id"];
+  string provider_address = 2 [json_name = "provider_address"];
+  bytes encrypted_data = 3 [json_name = "encrypted_data"];
+  bytes data_hash = 4 [json_name = "data_hash"];
+}
+
+message ValidateDataResponse {
+  Certificate certificate = 1;
+}
+
+// Certificate defines a certificate signed by an oracle who issued the certificate.
+message Certificate {
+  UnsignedCertificate unsigned_certificate = 1 [json_name = "unsigned_certificate"];
+  bytes signature = 2;
+}
+
+// UnsignedCertificate defines a certificate issued by an oracle as a result of data validation.
+message UnsignedCertificate {
+  string cid = 1;
+  string unique_id = 2 [json_name = "unique_id"];
+  string oracle_address = 3 [json_name = "oracle_address"];
+  uint64 deal_id = 4 [json_name = "deal_id"];
+  string provider_address = 5 [json_name = "provider_address"];
+  bytes data_hash = 6 [json_name = "data_hash"];
+}
```

Comment:
I think you can refer to these messages that are already defined in the panacea-core repo: https://github.com/medibloc/panacea-core/blob/main/proto/panacea/datadeal/v2/consent.proto.

Ideally, the best practice for proto files that are used by multi-repo projects is creating a separated git repo only for these proto files. But now, I guess we can simply refer to the panacea-core repo.

Reply:
Does that mean move the `panacea-core`'s proto to `third_party/proto/panacea` of panacea-oracle?

Reply:
I think that it's not a good practice to copy those proto files into the `third_party/`, even though we already have some google proto files in that directory :)
Unfortunately, Protocol Buffer doesn't have a built-in dependency management. There are some third-parties, but for simplicity, you can download dependencies whenever you run `make proto-gen`, similar as https://github.com/medibloc/panacea-core/blob/main/scripts/protocgen.sh#L9

Reply:
I've also tried this one: https://github.com/stormcat24/protodep. It worked well basically, but as I remember, it had some issues. I'm not sure whether the issue has been resolved or not.

Also, `buf` supports dependency management, but I'm a newbie to it. https://docs.buf.build/tour/add-a-dependency

Reply:
I don't know if I understood it well, but this practice you mentioned can be used to import proto files?
I need Certificate message in deal.proto of our panacea-oracle.

Reply:
> Does that mean move the panacea-core's proto to third_party/proto/panacea in panacea-oracle?
Let's go as you said. I was confused. The panacea-core also has some copied proto files from cosmos-sdk in its `third_party/proto/`. It would be okay for now, but let's consider either using `buf` with dependency management later or creating a separated git repo that contains only proto files.

Reply:
OK. I got it. Thank you!

Reply:
I added in `proto` files. But.. it's a lot of files. T_T

Reply:
We don't need to copied all proto files in panacea-core. I minimized it in https://github.com/medibloc/panacea-oracle/pull/81